internal/server/v2/parser.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package v2
+package v2
+
+import "fmt"
+
+// split a string by "->" and "<-"
+func split(s string) ([]string, error) {
+	if len(s) < 2 {
+		return nil, fmt.Errorf("invalid query string: %s", s)
```

Comment:
Wonder if we want to use more formal error syntax. For example:

status.Errorf(codes.InvalidArgument, ".....")

https://pkg.go.dev/google.golang.org/grpc/codes

Reply:
Done